test/Microsoft.ComponentDetection.Detectors.Tests/ConanLockComponentDetectorTests.cs

```diff
@@ -0,0 +1,271 @@
+    ""options"": ""someOptionsString"",
+    ""package_id"": ""packageIdOfLibAnotherLibrary2"",
+    ""prev"": ""someHashOfLibAnotherLibrary2"",
+    ""requires"": [
+     ""5""
+    ],
+    ""context"": ""host""
+    },
+   ""5"": {
+    ""ref"": ""libanotherlibrary3/4.5.6#someHashOfLibAnotherLibrary3"",
+    ""options"": """",
+    ""package_id"": ""packageIdOfLibAnotherLibrary3"",
+    ""prev"": ""someHashOfLibAnotherLibrary3"",
+    ""context"": ""host""
+    },
+   ""6"": {
+    ""ref"": ""libanotherlibrary4/5.6.7#someHashOfLibAnotherLibrary4"",
+    ""options"": ""someOptionsString"",
+    ""package_id"": ""packageIdOfLibAnotherLibrary4"",
+    ""prev"": ""someHashOfLibAnotherLibrary4"",
+    ""modified"": true,
+    ""context"": ""host""
+    }
+  },
+  ""revisions_enabled"": true
+ },
+ ""version"": ""0.4"",
+ ""profile_host"": ""someLongProfileHostSettingsString\n"",
+ ""profile_build"": ""someLongProfileBuildSettingsString\n""
+}
+";
+
+    private readonly string testConanLockNoDependenciesString = @"{
+ ""graph_lock"": {
+  ""nodes"": {
+   ""0"": {
+    ""ref"": ""MyConanProject/None"",
+    ""options"": ""SomeLongOptionsString"",
+    ""path"": ""../conanfile.py"",
+    ""context"": ""host""
+   }
+  },
+  ""revisions_enabled"": true
+ },
+ ""version"": ""0.4"",
+ ""profile_host"": ""someLongProfileHostSettingsString\n"",
+ ""profile_build"": ""someLongProfileBuildSettingsString\n""
+}
+";
+
+    [TestMethod]
+    public void TestConanLock_SerializeDeserialize()
+    {
+        var sampleConanLock = new ConanLock
+        {
+            Version = "0.4",
+            ProfileBuild = "someProfileBuild",
+            GraphLock = new ConanLockGraph
+            {
+                RevisionsEnabled = true,
+                Nodes = new Dictionary<string, ConanLockNode>()
+                {
+                    {
+                        "0", new ConanLockNode
+                        {
+                            Context = string.Empty,
+                            Modified = false,
+                            Options = string.Empty,
+                            PackageId = "SomePackageId",
+                            Path = "D:/SomePath",
+                            Previous = "SomeOtherId",
+                            Reference = string.Empty,
+                            Requires = new[] { "0" },
+                        }
+                    },
+                },
+            },
+        };
+        var jsonString = JsonSerializer.Serialize(sampleConanLock, new JsonSerializerOptions { DefaultIgnoreCondition = System.Text.Json.Serialization.JsonIgnoreCondition.WhenWritingDefault });
+        var expectedJsonString = @"{""version"":""0.4"",""profile_build"":""someProfileBuild"",""graph_lock"":{""revisions_enabled"":true,""nodes"":{""0"":{""context"":"""",""modified"":false,""options"":"""",""package_id"":""SomePackageId"",""path"":""D:/SomePath"",""prev"":""SomeOtherId"",""ref"":"""",""requires"":[""0""]}}}}";
```

Comment:
Are we just testing JSON deserialization? I'm not sure if this is necessary.

Reply:
I used this test to get the JSON serialization annotations to working state. May not add much value unless we are making changes to the annotations, in which case it can help avoid breaking the contract unexpectedly.

Reply:
Removed it.